types/mongoose/index.d.ts

```diff
@@ -502,7 +507,7 @@ declare module "mongoose" {
      * Returns a promise that resolves when done.
      * @param callback executed when all docs have been processed
      */
-    eachAsync(fn: (doc: T) => any, callback?: (err: any) => void): Promise<T>;
+    eachAsync(fn: (doc: T) => any, options: EachAsyncOptions, callback?: (err: any) => void): Promise<T>;
```

Comment:
This is a breaking change for users who weren't passing any options. There should probably be two overloads, one with `options` and one without.

Reply:
Thanks for reviewing @andy-ms , commited the overload function definition and added documentation for the fn parameter